common/workunit/workunit.cpp

```diff
@@ -172,232 +172,81 @@ void doDescheduleWorkkunit(char const * wuid)
+{
+    StringBuffer subgraphScopeName;
+    subgraphScopeName.append(_rootScope);
 
-class CConstGraphProgress : public CInterface, implements IConstWUGraphProgress
+    StatisticScopeType scopeType = SSTgraph;
+    StatsScopeId rootScopeId;
+    verifyex(rootScopeId.setScopeText(_rootScope));
+    collector.setown(createStatisticsGatherer(_creatorType, _creator, rootScopeId));
+}
+
+void CWuGraphStats::beforeDispose()
 {
-    class CWuGraphStats : public CInterfaceOf<IWUGraphStats>
+    Owned<IStatisticCollection> stats = collector->getResult();
+
+    MemoryBuffer compressed;
     {
-    public:
-        CWuGraphStats(CConstGraphProgress &_parent, StatisticCreatorType _creatorType, const char * _creator, const char * _rootScope, unsigned _id)
-            : parent(_parent), creatorType(_creatorType), creator(_creator), id(_id)
-        {
-            StringBuffer subgraphScopeName;
-            subgraphScopeName.append(_rootScope);
+        MemoryBuffer serialized;
+        serializeStatisticCollection(serialized, stats);
+        compressToBuffer(compressed, serialized.length(), serialized.toByteArray());
+    }
 
-            StatisticScopeType scopeType = SSTgraph;
-            StatsScopeId rootScopeId;
-            verifyex(rootScopeId.setScopeText(_rootScope));
-            collector.setown(createStatisticsGatherer(_creatorType, _creator, rootScopeId));
-        }
-        virtual void beforeDispose()
-        {
-            Owned<IStatisticCollection> stats = collector->getResult();
+    unsigned minActivity = 0;
+    unsigned maxActivity = 0;
+    stats->getMinMaxActivity(minActivity, maxActivity);
 
-            MemoryBuffer compressed;
-            {
-                MemoryBuffer serialized;
-                serializeStatisticCollection(serialized, stats);
-                compressToBuffer(compressed, serialized.length(), serialized.toByteArray());
-            }
+    StringBuffer tag;
+    tag.append("sg").append(id);
 
-            unsigned minActivity = 0;
-            unsigned maxActivity = 0;
-            stats->getMinMaxActivity(minActivity, maxActivity);
-            parent.setSubgraphStats(creatorType, creator, id, compressed, minActivity, maxActivity);
-        }
-        virtual IStatisticGatherer & queryStatsBuilder()
-        {
-            return *collector;
-        }
+    IPropertyTree * subgraph = createPTree(tag);
+    subgraph->setProp("@c", queryCreatorTypeName(creatorType));
+    subgraph->setProp("@creator", creator);
+    subgraph->setPropInt("@minActivity", minActivity);
+    subgraph->setPropInt("@maxActivity", maxActivity);
 
-    protected:
-        CConstGraphProgress &parent;
-        Owned<IStatisticGatherer> collector;
-        StringAttr creator;
-        StatisticCreatorType creatorType;
-        unsigned id;
-    };
+    //Replace the particular subgraph statistics added by this creator
+    tag.append("[@creator='").append(creator).append("']");
```

Comment:
I assume this must work, as it looks like this is just relocated code..
But am wondering how/where the 'sgXX' element that this code is trying to access gets set?
code is : 
```
    StringBuffer tag;
    tag.append("sg").append(id);
....
    //Replace the particular subgraph statistics added by this creator
    tag.append("[@creator='").append(creator).append("']");

    subgraph = progress->setPropTree(tag, subgraph);
```

For this qualified setPropTree to succeed, the sg with this creator must exist already.

@ghalliday - this is existing code, can you verify this works? As I say, I assume so, can you point out where the matching element is created before this call to update ?